mypy/checker.py

```diff
@@ -1076,7 +1076,8 @@ def check_assignment(self, lvalue: Node, rvalue: Node, infer_lvalue_type: bool =
                 elif (is_literal_none(rvalue) and
                         isinstance(lvalue, NameExpr) and
                         isinstance(lvalue.node, Var) and
-                        lvalue.node.is_initialized_in_class):
+                        lvalue.node.is_initialized_in_class and
+                        not self.options.fast_parser):
```

Comment:
This isn't correct -- it disallows the following (when run with the fast parser):
```py
class C:
    x = None # type: int
```

Instead, I think you'll need to add a flag for the new assignment syntax to `AssignmentStmt` and check for that here.